lambda/simplif.ml

```diff
@@ -430,6 +434,8 @@ let simplify_lets lam =
       count (bind_var bv v) l2;
       (* If v is unused, l1 will be removed, so don't count its variables *)
       if str = Strict || count_var v > 0 then count bv l1
+  | Lmutlet(_kind, v, _l1, l2) ->
+      count (bind_var bv v) l2
```

Comment:
If I read the code correctly, `Lmutlet` will never be substituted away by the `simplif` function below, even if the count is <=1. So why are we counting them? I think that we should count pure variables, not mutable variables, so this could be just `()` and same for the Lmutvar case above.